internals/cli/cmd_warnings_test.go

```diff
@@ -271,23 +271,23 @@ func (s *warningSuite) TestLastWarningTimestamp(c *check.C) {
 	err := os.WriteFile(newWarnPath, []byte("invalid JSON"), 0755)
 	c.Assert(err, check.IsNil)
 
-	rest, err := cli.Parser(cli.Client()).ParseArgs([]string{"okay"})
+	rest, err := cli.ParserForTest().ParseArgs([]string{"okay"})
 	c.Assert(err.Error(), check.Equals, "cannot decode timestamp file: invalid character 'i' looking for beginning of value")
 	c.Check(rest, check.HasLen, 1)
 
 	// Insert extra data after JSON
 	err = os.WriteFile(newWarnPath, []byte("{}extra"), 0755)
 	c.Assert(err, check.IsNil)
 
-	rest, err = cli.Parser(cli.Client()).ParseArgs([]string{"okay"})
+	rest, err = cli.ParserForTest().ParseArgs([]string{"okay"})
 	c.Assert(err.Error(), check.Equals, "spurious extra data in timestamp file")
 	c.Check(rest, check.HasLen, 1)
 
 	// Make open() fail
 	err = os.Chmod(newWarnPath, 0055)
 	c.Assert(err, check.IsNil)
 
-	rest, err = cli.Parser(cli.Client()).ParseArgs([]string{"okay"})
+	rest, err = cli.ParserForTest().ParseArgs([]string{"okay"})
```

Comment:
Btw, if it helps reviewing, I could separate out this big mechanical change (`cli.ParserForTest()`) into a separate PR to remove the noise from this PR. Let me know if you'd prefer that.

Reply:
Thanks, but that's probably more trouble than it's worth. It's not hard to scan down through a bunch of `ParserForTest()` lines.